### DISCUSSION TITLE
Variables: option to avoid setting "All" as default

### DISCUSSION THREAD

**Discussion by AntoineThebaud** [perses/perses#994] [Ideas]
Problem faced today with Grafana: sometimes you want to enable the "All" value for a variable, but you wouldn't want it to be the default value when
1. first loading the dashboard,
2. after changing a "parent" variable value (if the child value is not associated to the newly-selected parent value),

because in some cases the list of objects is very long and it's not relevant to select them all by default, which leads to long & useless loading time.

Grafana allows you to save a dashboard with the current-selected variables values as defaults, which solves point 1, but doesn't help in case 2. Therefore we could imagine having a checkbox to tell Perses if "All" should be put as default, or if it should be excluded, in this case it would select the 1rst value available in the dropdown (as currently for Grafana dashboards without the All option activated).

**Comment by sjcobb** [comment]
@foxbat07 and I were reading through this one again the other day and he thinks it's a good idea and is going to think about it when he creates mockups for saving default template variables and time range

**Comment by foxbat07** [comment]
@AntoineThebaud I think first one will get solved quickly once we have the ability to save variable defaults. if all is not feasible, we have talked about default to the first one while giving the user an alert of some sort to let them know whats happening and why.

For the second one - I think this will take some out of box thinking to better show relationship between variables in the UI itself and not the edit variable panel. I dont think this is a priority right now afaik. But will keep in mind!

**Comment by celian-garcia** [comment]
Hello, I'm just reading this discussion and realizing that I implemented recently something related. #1312 

The default selection for lists (when nothing is set as default value) is now the first value of the list. Even if `All` is allowed.
It means that the `All` value cannot be set "accidentally" if we modify another. @AntoineThebaud don't you think it solves completely the problem?

To me we could also benefit extending a bit more the "save current as defaults". I personally would like a feature where, when I save, I can select the variable that I want to save as default or not. Seeing which variable is saved but with which value. WDYT?

**Comment by AntoineThebaud** [comment]
> Hello, I'm just reading this discussion and realizing that I implemented recently something related. #1312
> 
> The default selection for lists (when nothing is set as default value) is now the first value of the list. Even if `All` is allowed. It means that the `All` value cannot be set "accidentally" if we modify another. @AntoineThebaud don't you think it solves completely the problem?
> 
> To me we could also benefit extending a bit more the "save current as defaults". I personally would like a feature where, when I save, I can select the variable that I want to save as default or not. Seeing which variable is saved but with which value. WDYT?

Wow I missed to reply here, sorry :( Indeed that solved it, and now 2 years later we have people asking for the previous behavior :D https://github.com/perses/perses/issues/3019. Let's close this discussion and continue the discussion there.
